indico/MaKaC/common/output.py

```diff
@@ -494,41 +495,25 @@ def _generate_contrib_location_and_time(self, contrib, out):
             out.closeTag('datafield')
 
     def _generate_contrib_people(self, contrib, out, subcontrib=None):
-        aList = list(set(contrib.primary_authors) | set(contrib.secondary_authors))
         sList = list(subcontrib.person_links) if subcontrib else list(contrib.speakers)
-        list_ = {}
-        auth = list(contrib.primary_authors)
-        if auth:
-            auth = auth[0]
-            list_[auth] = ['Primary Author']
-        for user in aList:
-            if user in list_:
-                if user != auth:
-                    list_[user].append('Author')
-            else:
-                list_[user] = ['Author']
+        users = defaultdict(list)
```

Comment:
I would use `persons` for clarity. We are dealing with `User` not with `PersonLink` here.